tests/PHPStan/Rules/Functions/StrvalFamilyParametersRuleTest.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleLevelHelper;
+use PHPStan\Testing\RuleTestCase;
+
+/**
+ * @extends RuleTestCase<StrvalFamilyParametersRule>
+ */
+class StrvalFamilyParametersRuleTest extends RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		$broker = $this->createReflectionProvider();
+		return new StrvalFamilyParametersRule(new RuleLevelHelper($broker, true, false, true, false));
+	}
+
+	public function testRule(): void
+	{
+		$this->analyse([__DIR__ . '/data/strval.php'], [
+			[
+				'Parameter #1 $value of function strval expects string|int|float|bool|resource|Stringable|null, array{} given.',
+				19,
+			],
+			[
+				'Parameter #1 $value of function strval expects string|int|float|bool|resource|Stringable|null, stdClass given.',
+				23,
+			],
+			[
+				'Parameter #1 $value of function intval expects string|int|float|bool|resource|null, stdClass given.',
+				35,
+			],
+			[
+				'Parameter #1 $value of function floatval expects string|int|float|bool|resource|null, stdClass given.',
+				44,
+			],
+			[
+				'Parameter #1 $value of function floatval expects string|int|float|bool|resource|null, resource given.',
```

Comment:
[As mentioned](https://github.com/phpstan/phpstan/issues/6560#issue-1123920440) I am unsure if this error is actually intentional. PHP would be fine with it. I did a quick check and changing https://github.com/phpstan/phpstan-src/blob/4dd6b692ef9224725764a6daec1b680c9eb84e03/src/Type/ResourceType.php#L57 to `FloatType` would resolve this and introduce no errors in the tests, but I didn't want to change it without asking first.